sonic-slave-buster/cross-build-arm-python-reqirements.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+PRE_BUILT_TARGET_DIR=$1
+
+declare -a pkgs=("grpcio==1.38.0" "grpcio-tools==1.20.0" "m2crypto==0.36.0" "bitarray==1.5.3" "lxml==4.6.3")
```

Comment:
>pkgs [](http://example.com/codeflow?start=11&length=4)

What is the motivation of this script? It use completely new way to handle dependencies and versions, and it will challenge the design [https://github.com/Azure/SONiC/pull/684](https://github.com/Azure/SONiC/pull/684) (@xumia to check).

If the purpose is to cross compile some pypi packages with c++ code, will this answer help [https://stackoverflow.com/a/61019582/2514803](https://stackoverflow.com/a/61019582/2514803)?

We prefer the ways to manage dependencies and versions, such as
1. apt-get install
2. pip install
3. wget/curl
4. git clone

Reply:
This script cross-compiles dockers' python packages heaviest (by building time) dependencies. 
Dockers install a number of python packages which in turn install their dependencies. Installing the dependencies takes a big amount of time using qemu emulation. And also all dockers that use these heavy dependencies build them from scratch again and again and waste the building time.
I wanted to prebuild these dependencies using cross-compiling - much faster than using qemu emulation. And copy them to the docker image (through Dockerfile.j2) to be used (without building them) by the python packages installed in the docker image. This approach saves a lot of building time.
So what is wrong that I use specific dependencies version ?
But dockers' Dockerfile.j2 also contains specific python packages versions, like this:
RUN pip3 install thrift==0.13.0
So these are specific python dependencies versions of the specific python packages versions.
Please instruct me how can I improve this approach.
Thanks.

Reply:
We only support pip3 install now, but the constraint option (-c <constraints.txt>) is also supported by pip3 download.

Reply:
@xumia has some idea how to implement reproducible build for `pip download`. I can close this comment.